modules/api/src/main/java/org/apache/fluo/api/config/SimpleConfiguration.java

```diff
@@ -335,4 +334,35 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
     ByteArrayInputStream bais = new ByteArrayInputStream(data);
     load(bais);
   }
+
+  private String stream2String(InputStream in) {
+    try {
+      ByteArrayOutputStream result = new ByteArrayOutputStream();
+      byte[] buffer = new byte[4096];
+      int length;
+      while ((length = in.read(buffer)) != -1) {
+        result.write(buffer, 0, length);
+      }
+
+      return result.toString(StandardCharsets.UTF_8.name());
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  /*
+   * Commons config 1 was used previously to implement this class. Commons config 1 required
+   * escaping interpolation. This escaping is no longer required with commmons config 2. If
+   * interpolation is escaped, then this API behaves differently. This function suppresses escaped
+   * interpolation in order to maintain behavior for reading.
+   */
+  private Reader cleanUp(InputStream in) {
+    return new StringReader(stream2String(in).replace("\\${", "${"));
```

Comment:
Current versions of Fluo rely on interpolation and document it in the default configs.  With CC1 this escaping was needed to make it work.  Without this things that used to work fail in odd and hard to debug way that will waste peoples time. I think something must be done to avoid that.

Reply:
I think that's a reasonable approach... I'm just not sure how far you want to take it... there may be other features of CC1 that changed in CC2 (or will change in future), and I'm trying to understand if you want to try to set a precedent here, or if this is a one-off thing.

Also, even if it's a one-off thing, is your intention to document the change in behavior and drop it eventually, or maintain this backwards-compatible behavior indefinitely?

Reply:
I changed this to throw an exception.